chain/chain/src/chain.rs

```diff
@@ -1409,11 +1409,20 @@ impl Chain {
             return;
         };
 
-        self.blocks_delay_tracker.record_optimistic_block_ready(block.height());
-
         let prev_block_hash = *block.prev_block_hash();
         let block_hash = *block.hash();
         let block_height = block.height();
+
+        self.blocks_delay_tracker.record_optimistic_block_ready(block_height);
+        if self.is_height_processed(block_height)? {
```

Comment:
Should this be a check for processed or received? 

side note:
 I wouldn't worry too much about forks / double signing, it's for optimistic block so worst case it'll just add a bit of delay